### PR TITLE
Splits project into UI and Core [Again]

### PR DESCRIPTION
**Pull request by JLLeitschuh**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,109 +1,147 @@
...
 
-jacocoTestReport {
-    reports {
-        xml.enabled = true
-        html.enabled = true
+    mainClassName = 'edu.wpi.grip.core.Main'
+
+    jar {
+        manifest {
+            attributes 'Implementation-Title': 'GRIP CORE', 'Implementation-Version': version, 'Main-Class': mainClassName
```
Comment (ThomasJClark) [inline build.gradle]: What is the point of the "Implementation-Title" attribute?

File: build.gradle [hunk 2]
```diff
@@ -1,109 +1,147 @@
...
-// End IDE setup
+project(":ui") {
+    apply plugin: 'java'
+    apply plugin: 'application'
+    apply from: 'http://dl.bintray.com/shemnon/javafx-gradle/8.1.1/javafx.plugin'
+
+    dependencies {
+        compile project(path: ':core', configuration: 'shadow')
+        compile group: 'org.controlsfx', name: 'controlsfx', version: '8.40.10'
+        compile group: 'org.apache.commons', name: 'commons-lang3', version: '3.4'
```
Comment (ThomasJClark) [inline build.gradle]: Apache Commons should probably be included in the core, even if it's not used there right now.

Comment (JLLeitschuh) [inline build.gradle]: If it needs to be moved there it can be added there later. What use could it have in the core?

File: build.gradle (hunk 1 above)
Comment (JLLeitschuh) [inline build.gradle]: Just following this: https://github.com/johnrengelman/shadow#configure-manifest-file
Do you know?

Comment (ThomasJClark) [inline build.gradle]: I'm not sure what you mean by "following" something.  The link you posted seems to be an example of how to set attributes in a jar, not guidelines on what attributes to put in a jar.

**Comment by JLLeitschuh** [comment]
@WPIRoboticsProjects/grip If no one has a comment on this PR in the next 12 hours I'm going to assume that there isn't a problem with it. I really don't like having to continuously rebase this PR up the file tree.

**Comment by ThomasJClark** [comment]
:-1: For one thing, there are conflicts with master

**Comment by JLLeitschuh** [comment]
I understand that.

**Comment by JLLeitschuh** [comment]
Okay, now it's good.

**Comment by ThomasJClark** [comment]
Once you fix those ping me on gitter. Do not merge this yourself, I'll do it

**Comment by JLLeitschuh** [comment]
Fix what?

**Comment by ThomasJClark** [comment]
The conflicts that you literally just fixed

**Pull request merged**